rasa/core/policies/rule_policy.py

```diff
@@ -783,14 +787,14 @@ def _find_action_from_rules(
                         f"Predicted loop '{active_loop_name}' by overwriting "
                         f"'{ACTION_LISTEN_NAME}' predicted by general rule."
                     )
-                    return active_loop_name, LOOP_RULES
+                    return active_loop_name, LOOP_RULES, entered_unhappy_path
 
                 # do not predict anything
                 predicted_action_name = None
 
             if DO_NOT_VALIDATE_LOOP in unhappy_path_conditions:
-                logger.debug("Added `FormValidation(False)` event.")
-                tracker.update(LoopInterrupted(True))
+                logger.debug("Entered unhappy path. Loop will be interrupted.")
```

Comment:
this is happening when we return from unhappy path to the loop
```suggestion
                logger.debug("Returning from unhappy path. Loop will be notified that it was interrupted.")
```